providers/swift/swift.go

```diff
@@ -262,7 +262,11 @@ func (c *Container) get(name string, headers swift.Headers, checkHash bool) (io.
 	if err != nil {
 		return nil, errors.Wrap(err, "open object")
 	}
-	return file, err
+
+	return objstore.ObjectSizerReadCloser{
+		ReadCloser: file,
+		Size:       file.Length,
```

Comment:
`file.Length()` returns the value of `Content-Length` response header if it is set, else it makes a HEAD call to the store